### PR TITLE
CDCG-54 - Create Web ReportsPatient Admin

### PR DESCRIPTION
**Pull request by IsaacGalindoBarbina**
### Feature / Bug Description
No existia un modulo de reportes donde se pueda visualizar una informacion detallada de las citas. Al menos conocer el nombre y celular del paciente

### Solution
Creacion de un modulo de reportes para poder consultar por medio de fechas que citas se encontraban dentro de un rango de fechas solicitadas.

### Notes
Add some notes

### Tickets
[CDCG-54](https://cdentalcaregroup.atlassian.net/browse/CDCG-54
CDCG-54 Module-ReportsPatient admin
)

### Screenshots / Screencasts
https://www.veed.io/view/8d682437-5063-434e-bd9c-a93a85b4291c?panel=share

[CDCG-54]: https://cdentalcaregroup.atlassian.net/browse/CDCG-54?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/data/reports/patientReport.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface ReportPatient{
+    name: string;
+    lastname: string;
```
Comment (imadiaz) [inline src/data/reports/patientReport.ts]: what is the difference between ReportPatient y Reports? looks like you have the same elements you can remove one of these classes

File: src/pages/reports/Reports.tsx [hunk 2]
```diff
@@ -1,16 +1,106 @@
...
+import SectionElement from '../components/SectionElement';
+import Strings from '../../utils/Strings';
+import { usePostReportsMutation } from '../../services/reportsService';
+import { handleErrorNotification } from '../../utils/Notifications';
+import moment from 'moment';
+
+const { Panel } = Collapse;
+
+interface Report {
+	name: string;
```
Comment (imadiaz) [inline src/pages/reports/Reports.tsx]: this is a duplicate class

File: src/pages/reports/Reports.tsx [hunk 3]
```diff
@@ -1,16 +1,106 @@
...
+	const [postReports] = usePostReportsMutation();
+
+	const handleConsultClick = async () => {
+		try {
+			const startDate = startedAt ? moment(startedAt, 'YYYY-MM-DD') : null;
+			const endDate = finishedAt ? moment(finishedAt, 'YYYY-MM-DD') : null;
+	
+			const result = await postReports({ startedAt: startDate, finishedAt: endDate }).unwrap();
+	
+			if (Array.isArray(result)) {
```
Comment (imadiaz) [inline src/pages/reports/Reports.tsx:22]: i think that its not necessary to validate this

File: src/data/reports/patientReport.ts (hunk 1 above)
Comment (IsaacGalindoBarbina) [reply src/data/reports/patientReport.ts]: I don't know why it's still visible but I already deleted it.

Comment (immanueldiaz) [reply src/data/reports/patientReport.ts]: If it's visible it means that you didn't update correctly your branch

File: src/data/reports/report.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export interface Reports {
-    name: string;
-    lastname: string;
-    primaryContact: string;
-    appointment: string;
+	name: string;
+	lastname: string;
+	primaryContact: string;
+	appointment: string;
```
Comment (imadiaz) [inline src/data/reports/report.ts]: if is the same structure for an user you can replace your class for
`Reports {
patient: Patient,
appointment: string
}
`

File: src/pages/reports/Reports.tsx (hunk 2 above)
Comment (imadiaz) [reply src/pages/reports/Reports.tsx]: +1

File: src/pages/reports/Reports.tsx [hunk 5]
```diff
@@ -1,16 +1,102 @@
...
+		}
+	};
+	
+	return (
+		<LayoutCard
+		  title={Strings.reports}
+		  isLoading={false}
+		  content={
+			<Collapse >
+			  <Panel header="Visualización de reportes" key="1">
```
Comment (imadiaz) [inline src/pages/reports/Reports.tsx]: please use a constants here

File: src/data/reports/report.ts (hunk 4 above)
Comment (IsaacGalindoBarbina) [reply src/data/reports/report.ts]: I have already made the changes

File: src/pages/reports/Reports.tsx (hunk 5 above)
Comment (IsaacGalindoBarbina) [reply src/pages/reports/Reports.tsx]: Okey

File: src/data/reports/patientReport.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+    primaryContact: string;
+  }
```
Comment (imadiaz) [inline src/data/reports/patientReport.ts]: if you already have Report, do we need this one?

Comment (IsaacGalindoBarbina) [reply src/data/reports/patientReport.ts]: I don't know but I have already deleted that, I have already done a pull and that file is not there. I don't know why you keep seeing it

Comment (IsaacGalindoBarbina) [reply src/data/reports/patientReport.ts]: Can u check if you look the same file

**Review by imadiaz — APPROVED** [review]

**Review by ThayliAreliVillegas112 — APPROVED** [review]

**Pull request merged**